### ISSUE TITLE
Automatically recognize and import Meteor component files

### ISSUE THREAD

**Issue by rhettlivingston** [Galooshi/import-js#295]
This is dependent on #230 as it is a side-effect type import.

Meteor modules supporting UI code commonly use the import for side-effects statement. Usually, just under all of the normal imports in "moduleXXX.js" you might find

```js
import './moduleXXX.http';
import './moduleXXX.css';
```

This is essentially an instruction to the build system to gather up this module's accompanying html and css files and send them to the client.

It would be helpful if a switch or other mechanism could be provided to instruct importjs to look for these accompanying files of the same name and automatically add imports for them when they are present.

This issue was created as a result of discussion in #289 that may contain other details and thoughts.

**Comment by rhettlivingston** [comment]
I'm starting to consider this one now. I reviewed the discussion in #289 under the light of all that has occurred since then and think we were nearing the solution when trotzig suggested an autoImport configuration hook.

I can't think of a reason for this routine to exist other than to supply "side-effect" type imports. Those are the only ones that should be impossible to determine by other means. Therefore, it would only have to return a list of ModuleSpecifiers to import. I believe that, in this case, those ModuleSpecifiers should all be absolute paths. The primary input would be pathToCurrentFile. The current working directory might be useful, but I'm starting to believe I've been overdependent on that.

I think it would only be applicable to "fix" operations?

I'm a bit iffy on the name. Maybe moduleSideEffectImports? That emphasises that it is per module and that it is only wanting these side-effect imports. In my case, it might look like
```js
moduleSideEffectImports({ pathToCurrentFile, config }: Object): Array<string> => {
  let basePath = '';
  if (pathToCurrentFile.endsWith('.js')) {
    basePath = pathToCurrentFile.slice(0, -3);
  } elseif (pathToCurrentFile.endsWith('.jsx')) {
    basePath = pathToCurrentFile.slice(0, -4);
  } else {
    return [];
  }
  const moduleSpecifiers = [];
  for (let ext of ['.html', '.css']) {
    const moduleSpecifier = `${basePath}${ext}`;
    if (fs.existsSync(path.join(config.workingDirectory, moduleSpecifier)) {
      moduleSpecifiers.push(moduleSpecifier);
  }
  return moduleSpecifiers;
}
```
The biggest problem I have with this is that many, perhaps most, but not all Meteor projects would want this. Meteor is a very flexible framework. Specifically, it supports multiple front ends including Blaze, React, and Angular. The above makes a lot of sense for Blaze and reflects the Meteor Guide's recommendations, but not so much for the others. I don't think it would hurt them though.

Bottom line is that I think we need to provide a way to turn it on and off separately from just not including the environment. Is that as simple as letting the user override this routine with their own routine? Or should there be some way for environments to create their own switches?

**Comment by trotzig** [comment]
Letting users disable it through an override would be good enough to begin with. I think your plan sounds like a good one. I like the name `moduleSideEffectImports`.
